File: sources/ocaml/fib.ml
(* http://www.balancer.ru/g/p1818588 *)

class fib n =
	object
		val v = n
		method value = match v with
			1 | 2 -> 1
			| i -> (new fib(i-1))#value + (new fib(i-2))#value
	end;;

let fib = new fib(40);;
print_int(fib#value);;

File: sources/rust/fib.rs
struct Fib {
	val: u32,
}

impl Fib {
	pub fn new(n: u32) -> Fib {
		Fib { val: n }
	}

	pub fn value(&self) -> u32 {
		if self.val <= 2 {
			return 1;
		}

		return Fib::new(self.val - 1).value() + Fib::new(self.val - 2).value();
	}

}

fn main() {
	for _ in 0..10 {
		println!("{}", Fib::new(40).value());
	}
}


